pages/cookbook/index.mdx

```diff
@@ -114,30 +114,216 @@ dump("The loop is over!");
+    map: map<Int, Int>; // «array» of Int values as a map of Ints to Ints
+    length: Int = 0;    // length of the «array», defaults to 0
+}
+
+// Extension mutation function for adding new entries
+extends mutates fun push(self: Array, item: Int) {
+    self.map.set(self.length, item); // set the entry (key-value pair)
+    self.length += 1;                // increase the length field
+}
+
+// Extension mutation function for deleting the last entry and returning its value
+extends mutates fun pop(self: Array): Int {
+    require(self.length > 0, "No items in the array to delete!");
+
+    // Note, that we use !! operator as we know for sure that the value would be there
+    let lastItem: Int = self.map.get(self.length - 1)!!;
+    self.map.set(self.length - 1, null); // delete the entry
+    self.length -= 1;                    // decrease the length field
+
+    return lastItem;
+}
+
+// Extension function for returning the last value
+extends fun getLast(self: Array): Int {
+    require(self.length > 0, "No items in the array!");
+
+    // Note, that we use !! operator as we know for sure that the value would be there
+    return self.map.get(self.length - 1)!!;
+}
+
+contract MapsAsArrays with Deployable {
+    // Persistent state variables
+    array: Array; // our «array», which uses the map
+
+    // Constructor (initialization) function of the contract
+    init() {
+        self.array = Array{map: emptyMap()}; // length defaults to 0
+    }
+
+    // Internal message receiver, which responds to a String message "push"
+    receive("push") {
+        // Add a new item and increase the counter
+        self.array.push(42);
+    }
+
+    // Internal message receiver, which responds to a String message "pop"
+    receive("pop") {
+        // Remove the last item and decrease the counter
+        self.array.pop();
+    }
+
+    // Internal message receiver, which responds to a String message "get_last"
+    receive("get_last") {
+        // Reply back with the latest item in the map if it exists, or raise an error
+        self.reply(self.array.getLast().toCoinsString().asComment());
+    }
+
+    // Getter function for obtaining the «array»
+    get fun map(): map<Int, Int> {
+        return self.array.map;
+    }
+
+    // Getter function for obtaining the current length of the «array»
+    get fun length(): Int {
+        return self.array.length;
+    }
+}
+```
+
+### How to make a circular buffer with a map
+
+```tact
+import "@stdlib/deploy"; // for Deployable trait
+
+// This contract records the last 5 timestamps of when "timer" message was received
+contract CircularBuffer with Deployable {
+    // Declare a compile-time constant upper bound for our map representing a buffer
+    const MaxBufferSize: Int = 5;
+
+    // Persistent state variables
+    buffer: map<Int, Int>; // this is our buffer implemented with a map
+    bufferLength: Int as uint8 = 0;
+    bufferStart: Int as uint8 = 0; // current index into the buffer
```

Comment:
I really like the example you did for arrays as maps packed with length + extension functions, looks like we should encourage that coding style for buffers as well (or any other complex data structure with non-trivial operations), let's try refactoring the circular buffer implementation in the same style